src/celeritas/ext/RootUniquePtr.hh

```diff
@@ -12,35 +13,51 @@
 #include "celeritas_config.h"
 #include "corecel/Assert.hh"
 
-// Forward-declare ROOT; Expand as needed
-class TFile;
-class TTree;
-class TBranch;
-
 namespace celeritas
 {
-namespace detail
+//---------------------------------------------------------------------------//
+/*!
+ * Call `TObject->Write()` before deletion. Used by TFile and TTree writer
+ * classes.
+ */
+template<class T>
+struct RootWritableDeleter
 {
+    void operator()(T* ptr);
+};
+
 //---------------------------------------------------------------------------//
-//! Helper to prevent ROOT from propagating to downstream code.
+/*!
+ * Custom deleter to avoid propagating any dependency-specific implementation
+ * downstream the code.
+ */
 template<class T>
-struct RootDeleter
+struct ExternDeleter
 {
-    void operator()(T*) const;
+    void operator()(T* ptr);
 };
 
+//---------------------------------------------------------------------------//
+// Type aliases
 template<class T>
-using RootUniquePtr = std::unique_ptr<T, RootDeleter<T>>;
+using UPRootWritable = std::unique_ptr<T, RootWritableDeleter<T>>;
+template<class T>
+using UPExtern = std::unique_ptr<T, ExternDeleter<T>>;
 
 //---------------------------------------------------------------------------//
 #if !CELERITAS_USE_ROOT
 template<class T>
-inline void RootDeleter<T>::operator()(T*) const
+void RootWritableDeleter<T>::operator()(T*)
+{
+    CELER_NOT_CONFIGURED("ROOT");
+}
+
+template<class T>
+void ExternDeleter<T>::operator()(T*)
```

Comment:
The one hiccup to our naming this `Extern` is that we provide this ROOT-oriented default implementation and don't provide the template definition for use cases outside of ROOT. (And of course the filename is still RootUniquePtr.) However I think those changes can all be addressed in a future PR we discussed that unifies the "extern deleter" bits.